backend/api/Models/Property/PropertyFilterModel.cs

```diff
@@ -0,0 +1,315 @@
+            this.PredominateUseId = filter.GetIntNullValue(nameof(this.PredominateUseId));
+            this.FloorCount = filter.GetIntNullValue(nameof(this.FloorCount));
+            this.Tenancy = filter.GetStringValue(nameof(this.Tenancy));
+            this.MinRentableArea = filter.GetFloatNullValue(nameof(this.MinRentableArea)) ?? filter.GetFloatNullValue(nameof(this.MinLotArea));
+            this.MaxRentableArea = filter.GetFloatNullValue(nameof(this.MaxRentableArea)) ?? filter.GetFloatNullValue(nameof(this.MaxLotArea));
+        }
+        #endregion
+
+        #region Methods
+        public static explicit operator ParcelFilter(PropertyFilterModel model)
+        {
+            var parcel = new ParcelFilter
+            {
+                NELatitude = model.NELatitude,
+                NELongitude = model.NELongitude,
+                SWLatitude = model.SWLatitude,
+                SWLongitude = model.SWLongitude,
+
+                Address = model.Address,
+                StatusId = model.StatusId,
+                ClassificationId = model.ClassificationId,
+                MinLandArea = model.MinLandArea ?? model.MinLotArea,
+                MaxLandArea = model.MaxLandArea ?? model.MaxLotArea,
+                MinEstimatedValue = model.MinEstimatedValue,
+                MaxEstimatedValue = model.MaxEstimatedValue,
+                MinAssessedValue = model.MinAssessedValue,
+                MaxAssessedValue = model.MaxAssessedValue,
+
+                Agencies = model.Agencies,
+                Sort = model.Sort
+            };
+
+            return parcel;
+        }
+
+        public static explicit operator BuildingFilter(PropertyFilterModel model)
+        {
+            var parcel = new BuildingFilter
+            {
+                NELatitude = model.NELatitude,
+                NELongitude = model.NELongitude,
+                SWLatitude = model.SWLatitude,
+                SWLongitude = model.SWLongitude,
+
+                Address = model.Address,
+                ConstructionTypeId = model.ConstructionTypeId,
+                PredominateUseId = model.PredominateUseId,
+                FloorCount = model.FloorCount,
+                Tenancy = model.Tenancy,
+                MinRentableArea = model.MinRentableArea ?? model.MinLotArea,
+                MaxRentableArea = model.MaxRentableArea ?? model.MaxLotArea,
+                MinEstimatedValue = model.MinEstimatedValue,
+                MaxEstimatedValue = model.MaxEstimatedValue,
+                MinAssessedValue = model.MinAssessedValue,
+                MaxAssessedValue = model.MaxAssessedValue,
+
+                Agencies = model.Agencies,
+                Sort = model.Sort
+            };
+
+            return parcel;
+        }
+
+        /// <summary>
+        /// Determine if a valid filter was provided.
+        /// </summary>
+        /// <returns></returns>
+        public bool ValidFilter()
+        {
+            return !(!this.NELatitude.HasValue
```

Comment:
unless I'm missing something the negations could mostly be removed and all of these switch to ||?

Reply:
You are probably right lol!  If any of them have values that it's a valid filter.